### DISCUSSION TITLE
(JEKYLL)Images Not Showing/Can't Remove Text

### DISCUSSION THREAD

**Discussion by ghost** [community/community#23717] [Other Feature Feedback, Questions, & Ideas]
<p>Hello Community,</p>
<p>For some reason my images are showing in the preview, but not live. The path is correct as images are showing in the background correctly on live site. Also I have text on the live site that isn’t supposed to be there as there is other text written locally in it’s place and not showing that text. I’m wondering if these 2 issues are cache related?</p>

**Comment by otter-computer** [comment]
<p>Hi <a class="mention" href="/fintechntrusts2020">@fintechntrusts2020</a>, can you drop a link to the repo that hosts your site so we can take a look at what’s going wrong?</p>

**Comment by ghost** [comment]
<p>Absolutely! <a href="https://github.com/fintechntrusts2020/Replica-Economics" rel="noopener nofollow ugc">https://github.com/fintechntrusts2020/Replica-Economics</a></p>

**Comment by ghost** [comment]
<p>It’s been a week since I posted my inquiry and no one has responded. I’m having some issues figuring out why my pictures are not displaying on the live site, and why text will not update live on Github Pages. If you’re interested I’m willing to compensate you for some assistance in resolving these issues.</p>

**Comment by otter-computer** [comment]
<p>Thanks for posting that link! Could you give some more information about what you mean by “images are showing in the preview?” I can see the images are loading correctly on your site but I’m not quite sure where else you’re looking for these images. Are they not working when you load your site locally, or are you looking at these images somewhere else?</p>
<p>Can you share some screenshots/links of where these images aren’t showing up?</p>

**Comment by ghost** [comment]
<p>Hello Thomas, and thank you for responding, I will explain the following in the following images:</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/f/f80152d0e7d7b80b9fcef148deee9f97ba1d49a9.png" data-download-href="/uploads/short-url/znXfoiG3qVrKLVaifUME9mlm5ip.png?dl=1" title="Capture3" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/70776696/181078881-76fdb5bb-70f7-4b65-85de-06f8779828c7.png" alt="Capture3" data-base62-sha1="znXfoiG3qVrKLVaifUME9mlm5ip" width="690" height="225" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/f/f80152d0e7d7b80b9fcef148deee9f97ba1d49a9_2_690x225.png, https://aws1.discourse-cdn.com/github/optimized/2X/f/f80152d0e7d7b80b9fcef148deee9f97ba1d49a9_2_1035x337.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/f/f80152d0e7d7b80b9fcef148deee9f97ba1d49a9.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/f/f80152d0e7d7b80b9fcef148deee9f97ba1d49a9_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Capture3</span><span class="informations">1360×445 208 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>In the screen above(which is the live site), the link appears to be broken, however in the image below there doesn’t appear to be anything wrong with the code.</p>

**Comment by ghost** [comment]
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/a/a130da2f4c4cab21d88c21825b51aee087db5c97.png" data-download-href="/uploads/short-url/mZXxn8Q55kb0viwFti0386oJ2vR.png?dl=1" title="Capture2" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/70776696/181078882-e2331e76-1643-4310-8466-d2ed55ecd554.png" alt="Capture2" data-base62-sha1="mZXxn8Q55kb0viwFti0386oJ2vR" width="690" height="112" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/a/a130da2f4c4cab21d88c21825b51aee087db5c97_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Capture2</span><span class="informations">1124×183 11.5 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by ghost** [comment]
<p>Below is the preview:</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/4/45d849da6615ee52fd6a71d86df16b343d2bc083.png" data-download-href="/uploads/short-url/9XSlanNgwuGxVPfRAeveiYgbA5l.png?dl=1" title="Capture4" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/70776696/181078883-2a1a3eab-867e-4fae-9992-0b6df823dbec.png" alt="Capture4" data-base62-sha1="9XSlanNgwuGxVPfRAeveiYgbA5l" width="690" height="449" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/4/45d849da6615ee52fd6a71d86df16b343d2bc083_2_690x449.png, https://aws1.discourse-cdn.com/github/optimized/2X/4/45d849da6615ee52fd6a71d86df16b343d2bc083_2_1035x673.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/4/45d849da6615ee52fd6a71d86df16b343d2bc083_2_1380x898.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/4/45d849da6615ee52fd6a71d86df16b343d2bc083_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Capture4</span><span class="informations">1532×998 100 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by ghost** [comment]
<p>Lastly, I’m not sure why this text hasn’t been updated and displayed correctly:</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/4/48fff58e3a548db1856de6cfd4ff0082b69dd39e.jpeg" data-download-href="/uploads/short-url/apMLBVMkRWVGyNY5i8rUCLAEl2K.jpeg?dl=1" title="Capture5.PNG" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/70776696/181078887-b78b24b4-7fa2-44fd-903a-55d32be16a7e.jpeg" alt="Capture5.PNG" data-base62-sha1="apMLBVMkRWVGyNY5i8rUCLAEl2K" width="690" height="328" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/4/48fff58e3a548db1856de6cfd4ff0082b69dd39e_2_690x328.jpeg, https://aws1.discourse-cdn.com/github/optimized/2X/4/48fff58e3a548db1856de6cfd4ff0082b69dd39e_2_1035x492.jpeg 1.5x, https://aws1.discourse-cdn.com/github/original/2X/4/48fff58e3a548db1856de6cfd4ff0082b69dd39e.jpeg 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/4/48fff58e3a548db1856de6cfd4ff0082b69dd39e_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Capture5.PNG</span><span class="informations">1098×523 109 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by ghost** [comment]
<p>The highlighted area should actually display the text “Objective” via the layout as shown below:</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/5/57cd45e9c81d3ec326b62dd509c09105ae89f187.png" data-download-href="/uploads/short-url/cwJkNIr5h2h4FAPNvzjNZP8WfpJ.png?dl=1" title="Capture6" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/70776696/181078889-f89ee526-b88c-4b4b-9be9-deb721909e65.png" alt="Capture6" data-base62-sha1="cwJkNIr5h2h4FAPNvzjNZP8WfpJ" width="690" height="399" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/5/57cd45e9c81d3ec326b62dd509c09105ae89f187_2_690x399.png, https://aws1.discourse-cdn.com/github/optimized/2X/5/57cd45e9c81d3ec326b62dd509c09105ae89f187_2_1035x598.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/5/57cd45e9c81d3ec326b62dd509c09105ae89f187.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/5/57cd45e9c81d3ec326b62dd509c09105ae89f187_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Capture6</span><span class="informations">1125×651 649 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by MichaelCurrin** [comment]
<p>Hi.</p>
<p>The text part is easy. Check out your layouts. You use the Mission twice. Once as mission/index.html using <code>default</code> layout and again as your demo post using <code>post</code> layout. Neither have a subtitle in them. So add <code>{{ subtitle }}</code> to the <code>post</code> layout like you do on <code>page</code> layout.</p>
<p>And you’ll probably want to turn mission/index.html to use page and not default layout.</p>

**Comment by MichaelCurrin** [comment]
<h2>Problem</h2>
<p>The reason your image is broken is because of your relative path.</p>
<p>Your overview page on the public URL here</p>
<p><a href="http://www.replica-eco.net/project/replica-overview" class="onebox" target="_blank" rel="noopener nofollow ugc">http://www.replica-eco.net/project/replica-overview</a></p>
<p>use a <code>frame</code> to load content from</p>
<aside class="onebox allowlistedgeneric">
  <header class="source">
      
      <a href="https://fintechntrusts2020.github.io/Replica-Economics/project/replica-overview" target="_blank" rel="noopener nofollow ugc">fintechntrusts2020.github.io</a>
  </header>
  <article class="onebox-body">
    <div class="aspect-image" style="--aspect-ratio:689/460;"><img src="https://user-images.githubusercontent.com/18750745/181078891-47963545-6a29-441f-a823-0c566df88ebe.jpeg" class="thumbnail" width="689" height="460" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/0/0c10a19dae80a0adcadca022654307f172a8b149_2_689x460.jpeg, https://aws1.discourse-cdn.com/github/optimized/2X/0/0c10a19dae80a0adcadca022654307f172a8b149_2_1033x690.jpeg 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/0/0c10a19dae80a0adcadca022654307f172a8b149_2_1378x920.jpeg 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/0/0c10a19dae80a0adcadca022654307f172a8b149_2_10x10.png"></div>

<h3><a href="https://fintechntrusts2020.github.io/Replica-Economics/project/replica-overview" target="_blank" rel="noopener nofollow ugc">What is Replica Economics? – Replica Economics</a></h3>

<p>Find out why this BCH Token exists to help you.</p>


  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>And on that page looking at HTML source, you have</p>
<pre><code class="lang-auto">&lt;p&gt;&lt;img src="/images/replica_logo_2.jpeg" alt="" /&gt;&lt;/p&gt;
</code></pre>
<p>Which should rather be:</p>
<pre><code class="lang-auto">&lt;p&gt;&lt;img src="/Replica-Economics/images/replica_logo_2.jpeg" alt="" /&gt;&lt;/p&gt;
</code></pre>
<p>Use <code>relative_url</code>.</p>
<p>Test it:</p>
<p><a href="https://fintechntrusts2020.github.io/Replica-Economics/images/replica_logo_2.jpeg" rel="noopener nofollow ugc">fintechntrusts2020.github.io/Replica-Economics/images/replica_logo_2.jpeg</a></p>
<h2>Recommended simple fix</h2>
<p>Therefore you should edit this page <a href="https://github.com/fintechntrusts2020/Replica-Economics/blob/4d5d8064bc465790ca9692c9ea54bcbf1ccf2b14/_projects/replica_overview.md" rel="noopener nofollow ugc">https://github.com/fintechntrusts2020/Replica-Economics/blob/4d5d8064bc465790ca9692c9ea54bcbf1ccf2b14/_projects/replica_overview.md</a></p>
<p>Change</p>
<pre><code class="lang-auto">![](/images/replica_logo_2.jpeg)
</code></pre>
<p>to</p>
<pre><code class="lang-auto">![Replica logo]({{ '/images/replica_logo_2.jpeg' | relative_url }})
</code></pre>
<p>Note your view in your .md file will not preview correctly because of using Jekyll / Liquid, but the site will build correctly.</p>
<p>The initial text will show if you image is broken, to help you know where on the page it is trying to load.</p>
<p>Secondly, to make that work, you must also edit your config <a href="https://github.com/fintechntrusts2020/Replica-Economics/blob/4d5d8064bc465790ca9692c9ea54bcbf1ccf2b14/_config.yml" rel="noopener nofollow ugc">https://github.com/fintechntrusts2020/Replica-Economics/blob/4d5d8064bc465790ca9692c9ea54bcbf1ccf2b14/_config.yml</a></p>
<p>To have:</p>
<pre><code class="lang-auto">baseurl: /Replica-Economics
</code></pre>
<p>That might already be set by GH Pages, but it is good to be explicit. Especially for local builds for <a href="https://localhost:4000/Replica-Economics/" rel="noopener nofollow ugc">https://localhost:4000/Replica-Economics/</a> to work.</p>

**Comment by MichaelCurrin** [comment, marked as answer]
<h2>Recommended restructure</h2>
<p>After you have got the images fixed, I would recommend restructuring your site.</p>
<p>It is very unintuitive to design and debug a public URL which is just a frame that points to your github pages site.</p>
<p>Your site is also likely to suffer in search engine results. Here are SEO issues:</p>
<ul>
<li>Google and other crawlers may not be able to read your <code>frame</code> content. In fact a frame is very 1990s and you might get penalized for just even using a frame. (An <em>iframe</em> is more modern, but still not a good fit for you).</li>
<li>The page will take longer to load using frames because of two steps.</li>
<li>You are effectively duplicating content on your two sites, which will give you a negative score on search engines.</li>
<li>Your sitemap on your public site <a href="http://www.replica-eco.net/sitemap.xml" rel="noopener nofollow ugc">www.replica-eco.net/sitemap.xml</a> references your <a href="http://github.io" rel="noopener nofollow ugc">github.io</a> domain. This will cause errors or penalties because the domain is different. One way to fix this is to update config with <code>url: https://www.replica-eco.net</code> so your sitemap builds correctly.</li>
<li>Plus you also going to want to fix your SSL issue to add HTTPS support to <a href="http://www.replica-eco.net" rel="noopener nofollow ugc">www.replica-eco.net</a> to improve SEO. Currently it’s staying on HTTP and giving me a warning as insecure.</li>
</ul>
<p>I suggest you get rid of the frame approach. Combine your two sites into one. Use GH Pages to build and host your content and your public domain on top of the same site (with no frame) to make it friendly.</p>
<p>You can use GitHub Pages’ free functionality of a public custom domain. Go to your repo settings and choose to setup a custom domain. You already own and paid for <a href="http://www.replica-eco.net" rel="noopener nofollow ugc">www.replica-eco.net</a> so you just have to configure GH Pages to use it. For example I think you’ll need to create a <code>CNAME</code> file in your repo. You might need to find a tutorial on this in the GH Pages docs if you need more info.</p>
<hr>
<h3>Subpath</h3>
<p>Oh and if you go this route you might want to go from serving on <code>/Replica-Economics</code> to <code>/</code></p>
<p>You’ll need to revert to this:</p>
<pre><code class="lang-auto">baseurl: ''
</code></pre>
<p>And rename the repo itself from <code>Replica-Economics</code> to <code>fintechntrusts2020.github.io</code></p>
<p>This is a GH Pages thing which turns the domain from <code>fintechntrusts2020.github.io/Replica-Economics</code> to <code>fintechntrusts2020.github.io</code>.</p>

**Comment by ghost** [comment]
<p>Hello there Michael, thank you for the helping! With your advice I fixed the issue with the text. However, when following your recommended simple fix, I still only get a broken link even after the config application. Is there something that I’m missing? The change:</p>
<aside class="quote no-group" data-username="michaelcurrin" data-post="12" data-topic="134789">
<div class="title">
<div class="quote-controls"></div>
 michaelcurrin:</div>
<blockquote>
<p><code>![Replica logo]({{ '/images/replica_logo_2.jpeg' | relative_url }})</code></p>
</blockquote>
</aside>
<p>Is to take place at the replica_overview.md file correct? Or is that change to be made in html?</p>

**Comment by MichaelCurrin** [comment]
<p>That is markdown code so it must be in a .md file.</p>
<p>Or you must rewrite as HTML then put it in .html</p>
<pre><code class="lang-auto">&lt;img src="{{ '/images/replica_logo_2.jpeg' | relative_url }}" 
    alt="Replica logo" /&gt;
</code></pre>
